stream_win.c

```diff
@@ -46,78 +63,55 @@ typedef BOOL (WINAPI *get_overlapped_result_ex_func)(
     BOOL alertable);
 static get_overlapped_result_ex_func get_overlapped_result_ex;
 
-static BOOL WINAPI win7_get_overlapped_result_ex(
+static BOOL WINAPI get_overlapped_result_ex_impl(
     HANDLE file,
     LPOVERLAPPED olap,
     LPDWORD bytes,
     DWORD millis,
     BOOL alertable) {
 
-  while (true) {
-    if (GetOverlappedResult(file, olap, bytes, FALSE)) {
-      // Result is available
-      return TRUE;
-    }
+  DWORD waitReturnCode, err;
+
+  stream_debug( "Preparing to wait for maximum %ums\n", millis );
+  if ( millis != 0 ) {
 
-    ULONGLONG start = GetTickCount64();
-    if (SleepEx(millis, alertable) == WAIT_IO_COMPLETION) {
+    waitReturnCode = WaitForSingleObjectEx(olap->hEvent, millis, alertable);
+    switch (waitReturnCode)
+    {
+    case WAIT_OBJECT_0:
+      // Event is signaled, overlapped IO operation result should be available.
+      break;
+    case WAIT_IO_COMPLETION:
+      // WaitForSingleObjectEx returnes because the system added an I/O completion
+      // routine or an asynchronous procedure call (APC) to the thread queue.
       SetLastError(WAIT_IO_COMPLETION);
+      break;
+    case WAIT_TIMEOUT:
+      // We reached the maximum allowed wait time, the IO operation failed
+      // to complete in timely fashion.
+      SetLastError(WAIT_TIMEOUT);
       return FALSE;
-    }
-    ULONGLONG end = GetTickCount64();
-
-    if (millis != INFINITE) {
-      millis = (DWORD)((ULONGLONG)millis - (end - start));
-      if (millis <= 0) {
-        // Out of time
-        SetLastError(WAIT_TIMEOUT);
-        return FALSE;
-      }
-    }
-  }
-}
 
-static BOOL WINAPI probe_get_overlapped_result_ex(
```

Comment:
I'd prefer to keep using the real GetOverlappedResultEx function if is is available.  For testing purposes, how about changing this probe function to check for something like WATCHMAN_WIN7_COMPAT=1 in the environment?

If that is set then we can just set `get_overlapped_result_ex = win7_get_overlapped_result_ex`, otherwise just run the code as is below.

That gives us the possibility of configuring a build matrix in the CI:

https://www.appveyor.com/docs/build-configuration#build-matrix

We can have one configuration with this set and one without it set, thus testing both code paths.
We could also use the same environment variable to force this mode on in the python code too.

You get bonus points and gratitude if you can include the appropriate changes to our appveyor.yml file to enable this as part of this pull request!